### ISSUE TITLE
FocusableActionDetector class needs translucent behavior

### ISSUE THREAD

**Issue by Tusshu** [flutter/flutter#92768]
I need to logout user if he is inactive for certain time in Flutter Web Application. 

I could achieve this by wrapping my application in FocusableActionDetector class and reset a timer every time user interacts with the app. After period of inactivity, timer fires and logs user out. 

This is possible if FocusableActionDetector class had translucent behavior similar to that for the GestureDetector.

**Comment by maheshmnj** [comment]
> I need to logout user if he is inactive for certain time in Flutter Web Application.

>I could achieve this by wrapping my application in FocusableActionDetector class and reset a timer every time user interacts with the app. After period of inactivity, timer fires and logs user out.

@Tusshu  I am not sure How `FocusableActionDetector` is related to your use case can you please explain?

**Comment by Tusshu** [comment]
My assumption is that if the application is wrapped in a FocusableActionDetector then every user action signal (keyboard, mouse etc) sent by the engine to flutter application will be captured by the detector before the signal reaches the child widget. That way I can know that some user action is taking place and hence keep the user session alive.

I am not sure if my assumption is correct. I don't see any other way to detect if the user is inactive for certain period and hence log him out.

**Comment by Tusshu** [comment]
Yes, I am referring to HitTestBehavior.translucent for GestureDetector.

I tried GestureDetector in the default web app but GestureDetector does not detect tap on the button. It does detect the tap on the main screen. So My approach will not work.

There need to be some mechanism for the main application to intercept user action event from the engine before it is sent to widget.

**Comment by maheshmnj** [comment]
> There need to be some mechanism for the main application to intercept user action event from the engine before it is sent to widget.

@Tusshu thanks for the clarification.

**Comment by yjbanov** [comment]
This seems to be purely a framework feature request. Removing the web label until we know if any web engine work is required to make this work.
